### PR TITLE
Add `MedianOfTwoSortedArrays.java` new algorithm

### PR DESCRIPTION
**Pull request by Hardvan**
- [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Java/blob/master/CONTRIBUTING.md).
- [x] This pull request is all my own work -- I have not plagiarized it.
- [x] All filenames are in PascalCase.
- [x] All functions and variable names follow Java naming conventions.
- [x] All new algorithms have a URL in their comments that points to Wikipedia or other similar explanations.
- [x] All new code is formatted with `clang-format -i --style=file path/to/your/file.java`

### CODE REVIEW

**Comment by alxkm** [comment]
It looks like it's [leetcode](https://leetcode.com/problems/median-of-two-sorted-arrays/description/), so according to the rules of this repository it can't be added.

https://github.com/TheAlgorithms/Java/blob/master/CONTRIBUTING.md

What do you think @siriak

**Comment by Hardvan** [comment]
I understand the concern about it being a LeetCode problem. However, I believe this implementation touches on a valid algorithm that has practical applications. Given its relevance and utility in algorithm studies, I think it warrants consideration for inclusion in the repository.

**Comment by siriak** [comment]
@Hardvan could you provide some links that explain this algorithm and its applications outside of leetcode context? If no, we cannot include it as it's too specific and only appears in the context of coding competitions.

**Comment by Hardvan** [comment]
@siriak The [Algorithms/Python repository](https://github.com/TheAlgorithms/Python/blob/master/data_structures/arrays/median_two_array.py) includes the "Median of Two Sorted Arrays" algorithm as part of its directory, indicating it is recognized as a valid algorithm beyond coding competitions.

**Comment by Hardvan** [comment]
Conflicts resolved!

File: src/test/java/com/thealgorithms/divideandconquer/MedianOfTwoSortedArraysTest.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import org.junit.jupiter.api.Test;
+
+public class MedianOfTwoSortedArraysTest {
+
+    @Test
+    void testFindMedianSortedArrays() {
+        // Test case 1: Arrays of equal length
+        assertEquals(2.5, MedianOfTwoSortedArrays.findMedianSortedArrays(new int[] {1, 3}, new int[] {2, 4}));
```
Comment (siriak) [inline src/test/java/com/thealgorithms/divideandconquer/MedianOfTwoSortedArraysTest.java:12]: Please use parameterized tests instead

Comment (Hardvan) [reply src/test/java/com/thealgorithms/divideandconquer/MedianOfTwoSortedArraysTest.java:12]: Refactored!

**Comment by Hardvan** [comment]
I have refactored the test code to use Parameterized tests instead,

**Review by siriak — DISMISSED** [review]

**Pull request merged**